### DISCUSSION TITLE
Layer 1 MIT Expo Bitcoin attack-a-thon ideas

### DISCUSSION THREAD

**Discussion by willcl-ark** [bitcoin-dev-project/warnet#237] [Ideas]
Soliciting ideas for a L1 attack-a-thon in April.

## Ideas

Current ideas:

* deploy a network with old node versions vulnerable to bugs (e.g. inflation bug) and see them exploited
* pull off an eclipse attack

## Work needed

tbd

**Comment by 0xB10C** [comment]
A while ago, I thought about an addrman attack-a-thon. The goal is to fill a certain bucket (or all buckets) with fake/malicious addresses.  Tools like [addrman-observer](https://addrman.observer/?url=https://gist.githubusercontent.com/0xB10C/41aad0f94229172ee799e0948047266e/raw/d46f58f52c16b93857f8e72cba704cfc9f866d9d/mainnet.json) could show the participants the current addrman of a node.

However, I don't think there are any publicly disclosed addrman attacks on old versions. Not sure if doing an attack-a-thon on something that's currently not considered broken is worth it. Likely not too rewarding for the participants. If someone comes up with an attack, it's likely a vulnerability, though.

**Comment by 0xB10C** [comment]
In the 2019 Chaincode Residency, John had a "Break Bitcoin Core" challenge for us, where he disabled some policy limits and re-allowed old opcodes. See the commits at [2019-06-breaking-bitcoin-core](https://github.com/jnewbery/bitcoin/commits/2019-06-breaking-bitcoin-core/). That could be a source for inspiration.

**Reply by josibake** [reply]
Removing policy limits is a great one. Scenarios could highlight certain DoS vectors that the policy limits were protecting against. We could build a "policy-less" node and push the image to our github repo.

**Reply by pinheadmz** [reply]
We can use `acceptnonstdtxn=1` too.
